src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewList.cs

```diff
@@ -0,0 +1,691 @@
+			itemContainer.AllowDrop = true;
+
+			if (IsContentMode)
+			{
+				bool hasChildren = itemContainer.HasUnrealizedChildren || itemNode.HasChildren;
+				itemContainer.GlyphOpacity = hasChildren ? 1.0 : 0.0;
+				if (itemContainer.IsExpanded != itemNode.IsExpanded)
+				{
+					var dispatcher = new DispatcherHelper(this);
+					dispatcher.RunAsync(() =>
+							{
+								itemNode.IsExpanded = itemContainer.IsExpanded;
+							});
+				}
+			}
+			else
+			{
+				itemContainer.IsExpanded = itemNode.IsExpanded;
+				itemContainer.GlyphOpacity = itemNode.HasChildren ? 1.0 : 0.0;
+			}
+
+			//Set startup TemplateSettings properties
+			var templateSettings = itemContainer.TreeViewItemTemplateSettings;
+			templateSettings.ExpandedGlyphVisibility = itemNode.IsExpanded ? Visibility.Visible : Visibility.Collapsed;
+			templateSettings.CollapsedGlyphVisibility = !itemNode.IsExpanded ? Visibility.Visible : Visibility.Collapsed;
+
+			base.PrepareContainerForItemOverride(element, item);
+
+			if (selectionState != itemNode.SelectionState)
+			{
+				ListViewModel.UpdateSelection(itemNode, selectionState);
+			}
+		}
+
+		protected override DependencyObject GetContainerForItemOverride()
+		{
+			var targetItem = new TreeViewItem();
+			return targetItem;
+		}
+
+		// IFrameworkElementOverrides
+		protected override void OnApplyTemplate()
+		{
+			if (!m_itemsSourceAttached)
+			{
+				ItemsSource = ListViewModel;
+				IsItemClickEnabled = true;
+				m_itemsSourceAttached = true;
+			}
+
+			base.OnApplyTemplate();
+		}
+
+		// IUIElementOverrides
+		protected override AutomationPeer OnCreateAutomationPeer() =>
+			new TreeViewListAutomationPeer(this);
+
+		internal string GetDropTargetDropEffect()
+		{
+			if (string.IsNullOrEmpty(m_dropTargetDropEffectString))
+			{
+				UpdateDropTargetDropEffect(true, false, null);
+			}
+			return m_dropTargetDropEffectString;
+		}
+
+		internal void SetDraggedOverItem(TreeViewItem newDraggedOverItem)
+		{
+			m_draggedOverItem = newDraggedOverItem;
+		}
+
+		internal void UpdateDropTargetDropEffect(bool forceUpdate, bool isLeaving, TreeViewItem keyboardReorderedContainer)
+		{
+			//Preserve old value of string in case it's needed.
+			string oldValue = m_dropTargetDropEffectString;
+
+			TreeViewItem dragItem = null;
+			AutomationPeer dragItemPeer = null;
+
+			if (keyboardReorderedContainer != null)
+			{
+				dragItem = keyboardReorderedContainer;
+			}
+			else
+			{
+				var listItem = ContainerFromItem(m_draggedOverItem) as TreeViewItem;
+				if (listItem != null)
+				{
+					dragItem = listItem;
+				}
+			}
+
+			if (dragItem != null)
+			{
+				var dragItemString = "";
+				var itemBeforeInsertPositionString = "";
+				var itemAfterInsertPositionString = "";
+				var draggedOverString = "";
+
+				dragItemPeer = FrameworkElementAutomationPeer.FromElement(dragItem);
+				if (dragItemPeer != null)
+				{
+					dragItemString = dragItemPeer.GetName();//TODO: dragItemPeer.GetNameCore();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)